backend/controllers/mainController.js

```diff
@@ -0,0 +1,140 @@
+/*
+ * ====================================================================================================================
```

Comment:
All of your code should be 80 lines max. It makes it really hard to review in Github and any diff tool if not.